jx_sqlite/expressions.py

```diff
@@ -61,11 +61,15 @@ def to_sql(self, schema, not_null=False, boolean=False):
                 tempb = tempa.setdefault(schema.get_column_name(col), {})
                 tempb[json_type_to_sql_type[col.type]] = quote_column(col.es_column).sql
 
-    return wrap([
-        {"name": relative_field(cname, self.var), "sql": types, "nested_path": nested_path}
-        for nested_path, pairs in acc.items() for cname, types in pairs.items()
-    ])
-
+    cols = []
+    for nested_path, pairs in acc.items():
+        for cname, types in pairs.items():
+            if not startswith_field(cname, self.var):
```

Comment:
The if/else block is hinting there is a problem. Most if/else blocks can be removed with "better" design or typing. The jx-sqlite code-base is littered with these if/else blocks, and every one of them should give you concern. The question is when you should be least concerned: You can be least concerned if the decision does not leak far outside of a code block; maybe performing final formatting for output; or parsing irregular inputs. Internal decision code is bad: It complicates the code path, adds context to the data structures, plus there is at least one other block of code that must "undo" the context the decision code introduced.  

For this case, the decision code is is compensating for a previous problem: The `cname` is relative to the `nested_path`, which causes the `..b` problem.  Please change this code to not use `self.var`, we should be done with `self.var` after the first line that finds all the references `cols`.  Include the relative name with the `cols` so they can be used in the construction of `acc` and the final output.

    cols = {cname: c for cname, cs in schema.map_to_sql(self.var).items() for c in cs}

I suspect this might lead to other efficiencies, like replacing `schema.get_column_name(child_col)` with something simpler.